How to build recast libraries?

Hello, 

 how to build and install Recast as library. I dont want demo or examples. Only valid libs + devel files for other project.

Preferably without alfa build systems like premake ( pretty bad idea use ALPHA build system)

GCC 8.2.1
cmake line:

`/usr/bin/cmake /home/abuild/rpmbuild/BUILD/recastnavigation-1.0.0+20181016.56dbf4c/. '-GUnix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DCMAKE_INSTALL_LIBDIR:PATH=/usr/lib64 -DCMAKE_BUILD_TYPE=RelWithDebInfo '-DCMAKE_C_FLAGS=-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -DNDEBUG' '-DCMAKE_CXX_FLAGS=-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -DNDEBUG' '-DCMAKE_Fortran_FLAGS=-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -DNDEBUG' '-DCMAKE_EXE_LINKER_FLAGS=-Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' -DCMAKE_MODULE_LINKER_FLAGS=-Wl,--as-needed '-DCMAKE_SHARED_LINKER_FLAGS=-Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' -DLIB_SUFFIX=64 -DCMAKE_SKIP_RPATH:BOOL=ON -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=ON -DBUILD_STATIC_LIBS:BOOL=OFF -DCMAKE_COLOR_MAKEFILE:BOOL=OFF -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_MODULES_INSTALL_DIR=/usr/share/cmake/Modules -DRECASTNAVIGATION_DEMO=OFF -RECASTNAVIGATION_TESTS=OFF -DRECASTNAVIGATION_EXAMPLES=OFF -DRECASTNAVIGATION_STATIC=OFF ..`

always fails in Detour

```
9s] /home/abuild/rpmbuild/BUILD/recastnavigation-1.0.0+20181016.56dbf4c/DetourCrowd/Source/DetourProximityGrid.cpp:63: undefined reference to `dtFree(void*)'
[    9s] /usr/lib64/gcc/x86_64-suse-linux/8/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/DetourCrowd.dir/Source/DetourProximityGrid.cpp.o: in function `dtFreeProximityGrid(dtProximityGrid*)':
[    9s] /home/abuild/rpmbuild/BUILD/recastnavigation-1.0.0+20181016.56dbf4c/DetourCrowd/Source/DetourProximityGrid.cpp:39: undefined reference to `dtFree(void*)'
[    9s] collect2: error: ld returned 1 exit status
[    9s] make[2]: *** [DetourCrowd/CMakeFiles/DetourCrowd.dir/build.make:162: DetourCrowd/libDetourCrowd.so.1.0.0] Error 1
[    9s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/recastnavigation-1.0.0+20181016.56dbf4c/build'
[    9s] make[1]: *** [CMakeFiles/Makefile2:204: DetourCrowd/CMakeFiles/DetourCrowd.dir/all] Error 2
[   10s] [ 58%] Linking CXX shared library libDetour.so
[   10s] cd /home/abuild/rpmbuild/BUILD/recastnavigation-1.0.0+20181016.56dbf4c/build/Detour && /usr/bin/cmake -E cmake_link_script CMakeFiles/Detour.dir/link.txt --verbose=1
[   10s] /var/lib/build/ccache/bin/c++ -fPIC -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -DNDEBUG -O2 -g -DNDEBUG -Wl,--as-needed -Wl,--no-undefined -Wl,-z,now -shared -Wl,-soname,libDetour.so.1 -o libDetour.so.1.0.0 CMakeFiles/Detour.dir/Source/DetourAlloc.cpp.o CMakeFiles/Detour.dir/Source/DetourAssert.cpp.o CMakeFiles/Detour.dir/Source/DetourCommon.cpp.o CMakeFiles/Detour.dir/Source/DetourNavMesh.cpp.o CMakeFiles/Detour.dir/Source/DetourNavMeshBuilder.cpp.o CMakeFiles/Detour.dir/Source/DetourNavMeshQuery.cpp.o CMakeFiles/Detour.dir/Source/DetourNode.cpp.o 
[   10s] cd /home/abuild/rpmbuild/BUILD/recastnavigation-1.0.0+20181016.56dbf4c/build/Detour && /usr/bin/cmake -E cmake_symlink_library libDetour.so.1.0.0 libDetour.so.1 libDetour.so
[   10s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/recastnavigation-1.0.0+20181016.56dbf4c/build'
[   10s] [ 58%] Built target Detour
[   10s] make[1]: Leaving directory '/home/abuild/rpmbuild/BUILD/recastnavigation-1.0.0+20181016.56dbf4c/build'
[   10s] make: *** [Makefile:144: all] Error 2
[   10s] error: Bad exit status from /var/tmp/rpm-tmp.4ypIep (%build)
```

--- a/DebugUtils/CMakeLists.txt
+++ b/DebugUtils/CMakeLists.txt
@@ -11,6 +11,11 @@ else()
     add_library(DebugUtils SHARED ${SOURCES})
 endif()
 
+target_link_libraries(DebugUtils
+    Recast
+    Detour
+)
+
 set_target_properties(DebugUtils PROPERTIES
         SOVERSION ${SOVERSION}
         VERSION ${VERSION}

--- a/DetourCrowd/CMakeLists.txt
+++ b/DetourCrowd/CMakeLists.txt
@@ -9,6 +9,10 @@ else ()
     add_library(DetourCrowd SHARED ${SOURCES})
 endif ()
 
+target_link_libraries(DetourCrowd
+    Detour
+)
+
 set_target_properties(DetourCrowd PROPERTIES
         SOVERSION ${SOVERSION}
         VERSION ${VERSION}

--- a/DetourTileCache/CMakeLists.txt
+++ b/DetourTileCache/CMakeLists.txt
@@ -9,6 +9,10 @@ else ()
     add_library(DetourTileCache SHARED ${SOURCES})
 endif ()
 
+target_link_libraries(DetourTileCache
+    Detour
+)
+
 set_target_properties(DetourTileCache PROPERTIES
         SOVERSION ${SOVERSION}
         VERSION ${VERSION}

